### PR TITLE
Bug 1984942: ApplyClusterRoleBinding triggers boundsError when adding new subject

### PR DESCRIPTION
**Pull request by dobsonj**
https://bugzilla.redhat.com/show_bug.cgi?id=1984942
cc @openshift/storage

### CODE REVIEW

File: pkg/operator/resource/resourceapply/rbac.go [hunk 1]
```diff
@@ -80,7 +80,7 @@ func ApplyClusterRoleBinding(ctx context.Context, client rbacclientv1.ClusterRol
 
 	requiredCopy.RoleRef.APIGroup = rbacv1.GroupName
 	for i := range requiredCopy.Subjects {
-		if existingCopy.Subjects[i].Kind == "User" {
+		if requiredCopy.Subjects[i].Kind == "User" {
```
Comment (jsafrane) [inline pkg/operator/resource/resourceapply/rbac.go:83]: Can you please fix also the same bug in `ApplyRoleBinding`?

Comment (dobsonj) [reply pkg/operator/resource/resourceapply/rbac.go:83]: Oops, missed that, thanks. Done.

**Comment by jsafrane** [comment]
/lgtm
/assign @deads2k @sttts 
for approval

**Pull request merged**